### DISCUSSION TITLE
isRefetching result variable

### DISCUSSION THREAD

**Discussion by swain** [TanStack/query#1390] [Ideas]
It seems like the basic `useQuery` API should return an `isRefetching` variable.

This boolean would be `true` if the data is loading AND it is not the first load (on mount).

A common motivating example: wanting to manage UI around a "pull-to-refresh" behavior.

```jsx
const { data, isLoading, isRefetching, refetch } = useQuery(...)

return (
  <div>
    { isLoading && !isRefetching
      ? <div>Loading</div>
      : <div>
           {data}
           {
             isRefetching
               ? <div>Refetching</div>
               : <button onClick={() => refetch()}>Refresh</button>
            }
         </div>
      : 
  </div>
);
```

Seems simple enough to implement -- would a PR be accepted?

**Comment by TkDodo** [comment]
It does return `isFetching`, which is always true when a request is in flight, including the first time (when you are isLoading). So isRefetching is just isFetching && !isLoading
